index.bs

```diff
@@ -74,903 +60,162 @@ specification uses that specification and terminology.
-: <dfn attribute for="IncomingStream">readingAborted</dfn>
-:: The `readingAborted` attribute represents a promise that is [=resolved=]
-    when the a message from the remote side aborting the stream is received.
-    For QUIC, that message is a RST_STREAM frame. When the stream receives
-    this mesage, the user agent MUST run the following:
-      1. Let |stream| be the {{IncomingStream}} object for which the abort
-         message was received.
-      1. Let |transport| be the {{WebTransport}}, which the |stream| was created
-         from.
-      1. [=Resolve=] the promise with the resulting {{StreamAbortInfo}} with
-         the {{StreamAbortInfo/errorCode}} set to the value from the aborting
-         message from the remote side.
-
-## Methods ##  {#incoming-stream-methods}
-
-: <dfn method for="IncomingStream">abortReading()</dfn>
-:: A hard shutdown of the {{IncomingStream}}. It may be called regardless of
-   whether the {{IncomingStream}} was created by the local or remote peer. When
-   the `abortWriting` method is called, the user agent MUST run the following
-   steps:
-     1. Let |stream| be the {{IncomingStream}} object which is about to abort
-        reading.
-     1. Let |transport| be the {{WebTransport}}, which the |stream| was created
-        from.
-     1. Let |abortInfo| be the first argument.
-     1. Start the closing procedure by sending a message to the remote side
-        indicating that the stream has been aborted (for QUIC, this is a
-        STOP_SENDING frame) with its error code set to the value of
-        |abortInfo.errorCode|.
-
-: <dfn method for="IncomingStream">arrayBuffer()</dfn>
-:: A convenience method that asynchronously reads all the contents of
-   |IncomingStream.readable| and returns it as an ArrayBuffer.  This locks the
-   stream while reading, just as if |IncomingStream.readable.getReader()| were
-   used.
-
-# Interface `WebTransportStream` #  {#web-transport-stream}
-
-A collection of common attributes and methods of all streams.
+The {{CompressionStream}}(format) constructor, when invoked, must run these steps:
+    1. If <dfn for=CompressionStream>format</dfn> is unsupported in CompressionStream, then throw a TypeError.
+    1. Let *cs* be a new CompressionStream object.
+    1. Set *cs*'s *format* to <a for=CompressionStream>format</a>.
+    1. Let *startAlgorithm* be an algorithm that takes no arguments and returns nothing.
+    1. Let *transformAlgorithm* be an algorithm which takes a *chunk* argument and runs the <a>compress and enqueue a chunk</a> algorithm with *cs* and *chunk*.
+    1. Let *flushAlgorithm* be an algorithm which takes no argument and runs the <a for=CompressionStream>flush and enqueue</a> algorithm with *cs*.
+    1. Let *transform* be the result of calling <a abstract-op>CreateTransformStream</a>(*startAlgorithm*, *transformAlgorithm*, *flushAlgorithm*).
+    1. Set *cs*'s <a for=CompressionStream>transform</a> to *transform*.
```

Comment:
I think we shouldn't have `for=CompressionStream` here, because we want to link to where it is defined in GenericTransformStream. Sorry if I told you to add it :smile:.